source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -1323,142 +1323,161 @@ namespace RendererQml
-		}
+        if (!input->GetValue().empty() && Utils::isValidDate(input->GetValue()))
+        {
+            uiTextField->Property("text", Formatter() << Utils::GetDate(input->GetValue()) << ".toLocaleString(Qt.locale(\"en_US\"),"
+                << "\"" << StringDateFormat << "\""
+                << ")");
+        }
+
+        uiTextField->Property("validator", Formatter() << "RegExpValidator { regExp: " << DateRegex << "}");
+        uiTextField->Property("onFocusChanged", Formatter() << "{"
+            << "if(focus===true) inputMask=\"" << inputMask << "\";"
+            << "if(focus === false){ "
+            << "if(text === \"" << std::string(dateSeparator) + std::string(dateSeparator) << "\"){ inputMask = \"\" ; } "
+            << "}} ");
+        calendarTag->Property("onReleased", Formatter() << "{" << calendar_box_id << ".close(); " << uiTextFieldId << ".text=selectedDate.toLocaleString(Qt.locale(\"en_US\"),"
+            << "\"" << StringDateFormat << "\")}");
 
-		uiDateInput->Property("validator", Formatter() << "RegExpValidator { regExp: " << DateRegex << "}");
-		uiDateInput->Property("onFocusChanged", Formatter() << "{" << "if(focus===true) inputMask=\"" << inputMask << "\";" <<
-			"if(focus === false){ z=0;" << "if(text === \""<< std::string(dateSeparator) + std::string(dateSeparator) << "\"){ inputMask = \"\" ; } " <<"if( " << calendar_box_id << ".visible === true){ " << calendar_box_id << ".visible=false}}} ");
-        calendarTag->Property("onReleased", Formatter() << "{parent.visible=false; " << input->GetId() << ".text=selectedDate.toLocaleString(Qt.locale(\"en_US\")," << "\"" << StringDateFormat << "\")}");
+        calendarContentItemTag->Property("Component.onCompleted", "{ Qt.createQmlObject('" + calendarTag->ToString() + "', parent ,'calendar')}");
+        calendarBoxTag->Property("contentItem", calendarContentItemTag->ToString());
 
-		calendarBoxTag->Property("Component.onCompleted", "{ Qt.createQmlObject('" + calendarTag->ToString() + "'," + calendar_box_id + ",'calendar')}");
+        uiTextField->Property("placeholderText", Formatter() << (!input->GetPlaceholder().empty() ? input->GetPlaceholder() : "Select date") << " in " << Utils::ToLower(StringDateFormat), true);
 
-        uiDateInput->Property("placeholderText", Formatter() << (!input->GetPlaceholder().empty() ? input->GetPlaceholder() : "Select date") << " in " << Utils::ToLower(StringDateFormat), true);
-        uiDateInput->AddChild(calendarBoxTag);
+        auto uiDateInput = std::make_shared<QmlTag>("ComboBox");
+        uiDateInput->Property("id", input->GetId());
+        uiDateInput->Property("width", "parent.width");
+        uiDateInput->Property("background", uiTextField->ToString());
+        uiDateInput->Property("popup", calendarBoxTag->ToString());
+        uiDateInput->Property("indicator", iconTag->ToString());
```

Comment:
Placed the existing implementation (with slight changes) of the Input Date inside a combobox in order to make the popup display anchoring the textfield. Placing a popup separately would cover the textfield when opened. The same is done for Input Time as well